# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/css/css-transforms/transform-box/stroke-box-mutation-004.html
# Commit: e215151d2549
# Full Hash: e215151d2549581c23a1d3c8b53d8899f9776114
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2023-08-23 04:21:33
# Regressor Bug: 1819464
# File Overlap Count: 1
# Description:
#   Bug 1819464 - Part 5: Support stroke-box (for SVG layout). r=emilio
#   
#   We should avoid using the SVG shape elements, e.g. path, polygon, which may not
#   have simple bounds, and its stroke bbox contribution may depends on the
#   platform and the drawing backend if using Moz2D. So for now it'd be better to
# ==============================================================================

diff -r 803ea4d17810 -r e215151d2549 testing/web-platform/tests/css/css-transforms/transform-box/stroke-box-mutation-004.html
--- a/testing/web-platform/tests/css/css-transforms/transform-box/stroke-box-mutation-004.html	Tue Aug 22 19:49:04 2023 +0000
+++ b/testing/web-platform/tests/css/css-transforms/transform-box/stroke-box-mutation-004.html	Tue Aug 22 19:49:04 2023 +0000
@@ -1,32 +1,32 @@
 <!DOCTYPE html>
 <html class="reftest-wait">
 <title>transform-box: stroke-box, on group (&lt;g>), child shape mutated</title>
-<link rel="match" href="reference/svgbox-ref.html">
+<link rel="match" href="reference/svgbox-rect-ref.html">
 <link rel="help" href="https://drafts.csswg.org/css-transforms-1/#transform-box">
 <script src="/common/reftest-wait.js"></script>
 <script src="/common/rendering-utils.js"></script>
-<meta name="assert" content="This should display a C shape with the center of the top endpoint at 200,100."/>
+<meta name="assert" content="This should display a rect whose center of rotation is at (20, 0) of its stroke-box."/>
 <style>
 #target {
   transform-box: stroke-box;
-  transform-origin: 25px 0px;
+  transform-origin: 20px 0px;
   transform: rotate(90deg);
 }
-#target > path {
+#target > rect {
   fill: green;
   stroke: black;
-  stroke-width: 50;
+  stroke-width: 20;
 }
 </style>
 <svg width="400" height="300">
   <g id="target">
-    <path d="M200,100v50h100v-50"/>
+    <rect width="100" height="100" x="100" y="100"/>
   </g>
 </svg>
 <script>
   waitForAtLeastOneFrame().then(() => {
-    const path = document.querySelector('#target > path');
-    path.setAttribute('d', 'M200,100v100h100v-100');
+    const rect = document.querySelector('#target > rect');
+    rect.setAttribute('height', '50');
     takeScreenshot();
   });
 </script>